acceptance/cmd_server_test.go

```diff
@@ -0,0 +1,74 @@
+package acceptance_test
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/databricks/cli/internal/testcli"
+	"github.com/stretchr/testify/require"
+)
+
+func StartCmdServer(t *testing.T) *TestServer {
+	server := StartServer(t)
+	server.Handle("/", func(r *http.Request) (any, error) {
+		q := r.URL.Query()
+		args := strings.Split(q.Get("args"), " ")
+
+		var env map[string]string
+		require.NoError(t, json.Unmarshal([]byte(q.Get("env")), &env))
+
+		for key, val := range env {
+			defer Setenv(t, key, val)()
+		}
+
+		defer Chdir(t, q.Get("cwd"))()
+
+		c := testcli.NewRunner(t, context.Background(), args...)
```

Comment:
Use `r.Context()` to tear it down when the caller goes away.

Reply:
Updated in https://github.com/databricks/cli/pull/2184/commits/0da7dc6583a060768fa50fedf7b855af5e9329e0